src/app/static/src/app/utils.ts

```diff
@@ -348,3 +350,81 @@ export const getFormData = (file: File) => {
     formData.append("file", file)
     return formData
 }
+
+const transformPathToHash = (dataset: any) => {
+    Object.keys(dataset).map((key: string) => {
+        dataset[key] = {
+            hash: dataset[key].path.split("/")[1] || "",
+            filename: dataset[key].filename
+        }
+    })
+    return dataset
+}
+
+export const constructRehydrateProjectState = (rootState: RootState, data: ProjectRehydrateResultResponse) => {
+    const files = transformPathToHash({...data.state.datasets});
+
+    const modelCalibrate = {
+        ...rootState.modelCalibrate,
```

Comment:
I don't think we actually want to retain rootState modules' data here, do we? This root state originated with current project's root state, Using the initial state for each module might be safer. Maybe this might be part of why I'm seeing quota exceeded error? (Not sure!)